discovery/src/main/java/org/wildfly/extension/discovery/StaticProviderAddHandler.java

```diff
@@ -0,0 +1,116 @@
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.wildfly.extension.discovery;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.jboss.as.controller.AbstractAddStepHandler;
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.capability.RuntimeCapability;
+import org.jboss.as.controller.operations.global.WriteAttributeHandler;
+import org.jboss.as.controller.registry.Resource;
+import org.jboss.dmr.ModelNode;
+import org.wildfly.discovery.AttributeValue;
+import org.wildfly.discovery.ServiceURL;
+import org.wildfly.discovery.impl.MutableDiscoveryProvider;
+import org.wildfly.discovery.impl.StaticDiscoveryProvider;
+import org.wildfly.discovery.spi.DiscoveryProvider;
+
+/**
+ * @author <a href="mailto:david.lloyd@redhat.com">David M. Lloyd</a>
+ */
+class StaticProviderAddHandler extends AbstractAddStepHandler {
+    private static final StaticProviderAddHandler INSTANCE = new StaticProviderAddHandler();
+
+    static StaticProviderAddHandler getInstance() {
+        return INSTANCE;
+    }
+
+    private StaticProviderAddHandler() {
+        super(new Parameters().addAttribute(StaticProviderDefinition.SERVICES));
+    }
+
+    public void execute(final OperationContext context, final ModelNode operation) throws OperationFailedException {
+        context.registerCapability(
+            RuntimeCapability.Builder
+                .of(DiscoveryExtension.DISCOVERY_PROVIDER_CAPABILITY, true, new MutableDiscoveryProvider())
+                .build().fromBaseCapability(context.getCurrentAddressValue()));
+        super.execute(context, operation);
+    }
+
+    protected void recordCapabilitiesAndRequirements(final OperationContext context, final ModelNode operation, final Resource resource) throws OperationFailedException {
+        // no operation
+    }
+
+    protected void performRuntime(final OperationContext context, final ModelNode operation, final Resource resource) throws OperationFailedException {
+        modifyRegistrationModel(context, operation);
+    }
+
+    static void modifyRegistrationModel(OperationContext context, ModelNode op) throws OperationFailedException {
+        WriteAttributeHandler.INSTANCE.execute(context, op);
+        context.addStep(op, StaticProviderAddHandler::modifyRegistration, OperationContext.Stage.RUNTIME);
+    }
+
+    static void modifyRegistration(OperationContext context, ModelNode op) throws OperationFailedException {
+        final MutableDiscoveryProvider mutableDiscoveryProvider = context.getCapabilityRuntimeAPI(DiscoveryExtension.DISCOVERY_PROVIDER_CAPABILITY, context.getCurrentAddressValue(), MutableDiscoveryProvider.class);
+        if (op.hasDefined(DiscoveryExtension.SERVICES)) {
```

Comment:
Same comment re: 'model' vs 'op' that I had in AggregateProviderAddHandler applies here.